plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -358,4 +395,64 @@ static String getAttribute(ILaunchConfiguration configuration,
       return emptyMap;
     }
   }
+
+  @VisibleForTesting
+  Path getServiceAccountKeyPath() {
+    Preconditions.checkState(!projectSelector.getSelection().isEmpty());
+
+    String projectId = projectSelector.getSelectProjectId();
+    String filename = "app-engine-default-service-account-key-" //$NON-NLS-1$
```

Comment:
I wonder if we should make our own dot directory in home. What do you think?

Reply:
Can't really think of a good name, when considering this is not specific to our plugin and potentially may contain non-App Engine service accounts in the future.

I also feel unsure trying to hide it (dot directory) is better, as this is an important file.

Reply:
I think cluttering the user's home directory is a bad idea. dot directories are pretty standard for this sort of thing on Linux and not uncommon on Mac OS X. Not sure about Windows. Maybe we can ask around for suggestions.

Reply:
Crazy idea: what if we generated new keys for each local run and deleted them when we're done?

Reply:
> Crazy idea: what if we generated new keys for each local run and deleted them when we're done?

We will need give some thoughts to make this robust in case a run does not stop cleanly, not to litter the key space. There seems to be a hard limit on the number of keys (about 8~10) per service account.

Reply:
We could store this in the workspace plugin state location (e.g., `<workspace>/.metadata/.plugins/com.google.cloud.tools.eclipse.xxx`).  Or in the Eclipse configuration directory, which is shared across workspaces (`org.eclipse.core.runtime.Platform#getConfigurationLocation()` is the easiest accessor).

If we're thinking of a non-Eclipse location, typical OS-specific locations are:
  - on MacOS: `${user.home}/Library/Application Support/<appName>`
  - on Linux: `${XDG_DATA_HOME}/<appName>` since it's a data file (?) which defaults to `${user.home}/.local/share` according to [XDG](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html); alternatively we could use `${XDG_CONFIG_HOME}` if we think of this more as a config file
  - on Windows: `${LOCALAPPDATA}/<appName>` [source](http://environmentvariables.org/LocalAppData)

Reply:
Discussed this for a while with @elharo. The Eclipse configuration directory seems a workable option now.